src/utils/api.js

```diff
@@ -0,0 +1,34 @@
+import axios from 'axios';
+import { myFirebase } from '../firebase/firebase';
+
+let SOUP_API = '';
+
+if (process.env.NODE_ENV !== 'production') {
+  SOUP_API = 'http://localhost:5000';
+} else {
+  SOUP_API = 'https://souper-app-backend.herokuapp.com';
+}
+
+const api = axios.create({
+  baseURL: `${SOUP_API}/api/`,
+  headers: {
+    'Content-Type': 'application/json'
+  }
+});
+
+// Add a request interceptor
+api.interceptors.request.use(async config => {
+  config.headers['x-auth-token'] = await myFirebase
+    .auth()
+    .currentUser.getIdToken()
+    .then(async idToken => {
+      console.log(idToken);
+      return idToken;
+    })
+    .catch(error => {
```

Comment:
If the user is not authenticated we're going to print an error message in the login process, right? shouldn't we check if the user is logged first checking the state with redux thunk? and if there is an error shouldn't we logout inmmediately?.
About this one we supposed that it will be triggered when there is a 401 or 403 error but  maybe we can prevente it beforehand

Reply:
This is just to add the token in the header for the api requests. It needs some tidy up as its a bit rough currently, just getting the basic functionality in at the minute, and agreed that we should handle any errors accordingly. We should probably add an interceptor for the response and if theres a 401 we log them out.